### DISCUSSION TITLE
Custom values and Validation

### DISCUSSION THREAD

**Discussion by giosce** [publiccodeyml/publiccode.yml#135] [Ideas]
I need to change the list of "Scopes" values.
I proposed a PR but the comment is that this will create an invalid publiccode.yml.
First, I don't see where the software applies validation of values (I see it validating that mandatory fields are present).
In addition, I think that having these values in js files forces who needs different values to fork the project. I suggest (and I have done it in my fork) that an organization can provide the URL of its values as a property (at least as alternative).

**Comment by ruphy** [comment]
Hello @giosce, thanks for posting!

Could you explain what you're aiming for a little more clearly?
The scopes are currently defined in the following list, which is admittedly not perfect but serves as a decent starting point: https://yml.publiccode.tools/scope-list.html

With a PR to the standard, this could definitely be improved! Is this what you are looking for?

In addition, I'm not sure what the question about the js files means. Is it the editor you're referring to?

**Comment by bfabio** [comment]
Pasting the reply from the publiccode-editor repo, so we can continue here:

> **https://github.com/italia/publiccode-editor/pull/237**
>
> There is a reference validator at https://github.com/italia/publiccode-parser-go which checks for conformance to the publiccode.yml  Standard (https://yml.publiccode.tools/).
>
> As of now the Standard defines not only the keys in the YAML, but also the value they can hold, in some cases.

@giosce The aim is to provide a standard that makes sense internationally for any software, while also having a mechanism for each Country to provide their own Country-specific need.

Having said that, I think we can definitely come up with a one-size-fits-all taxonomy for scopes.

**Comment by giosce** [comment]
Writing here for now additional fields that Code For America would like to change:
Release Date - make it not mandatory: In many open-source, non-profit projects often there isn't an official release date, anyone is free to start using a software artifact at any time.
Scope - allow for an organization values: This is basically the Taxonomy of my organization (or as we are trying to do, the Civic Tech sector Taxonomy). As such, CivicTech values will be part of a CivicTech catalog search/filter. If AGID catalog has some items that don't match with CivicTech items, search in AGID catalog for example won't find "defence" project while AGID catalog won't find our "Homelessness" projects since "Homelessness" is not in AGID search interface.
Long Description - reduce mandatory length or make it optional
License - make it optional: when a project is in conceptual status maybe isn't clear yet which license to use
Maintenance/community/contact - the editor doesn't seem to check that is mandatory

In addition, another couple of thoughts:
Using the 2 letter codes for countries and languages is really not user friendly
I'm not able to see what adding a language does
The favicon shouldn't be "it"

**Reply by ruphy** [reply]
@giosce just answering to the LICENSE point, unfortunately any software without a license is closed source and can't even be used by third parties, so making it worse than proprietary.
Publiccode.yml is only meant for open source projects, as such we can't drop the requirements on the license. In this case, I suggest you to adopt a strong-copyleft license by default (such as AGPL), with the option of changing the license to a more liberal one later, any time you want.

**Reply by giosce** [reply]
Good explanation, thanks, I'll keep it in mind during my customization.
